Mathlib/Tactic/PNatToNat.lean

```diff
@@ -0,0 +1,61 @@
+/-
+Copyright (c) 2025 Vasilii Nesterov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Vasilii Nesterov
+-/
+import Mathlib.Data.PNat.Basic
+
+/-!
+# `pnat_to_nat`
+
+This file implements the `pnat_to_nat` tactic that shifts `PNat`s in the context to `Nat`.
+
+## Implementation details
+The implementation follows these steps:
+1. For each `x : PNat` in the context, add the hypothesis `0 < (↑x : ℕ)`.
+2. Translate arithmetic on `PNat` to `Nat` using the `pnat_to_nat_coe_simps` simp set.
+
+-/
+
+namespace Mathlib.Tactic.PNatToNat
+
+open private getElimNameInfo generalizeTargets generalizeVars from Lean.Elab.Tactic.Induction
+
+open Lean Meta Elab Tactic Qq Mathlib.Tactic
+
+/-- For each `x : PNat` in the context, add the hypothesis `0 < (↑x : ℕ)`. -/
+elab "pnat_positivity" : tactic => withMainContext do
+  let result ← (← getLCtx).foldlM (init := ← getMainGoal) fun g decl => do
+    let ⟨1, declType, declExpr⟩ ← inferTypeQ decl.toExpr | return g
+    let ~q(PNat) := declType | return g
+    let pf := q(PNat.pos $declExpr)
+    let (_, mvarIdNew) ← (← g.assert .anonymous q(0 < PNat.val $declExpr) pf).intro1P
```

Comment:
I wonder if it is worth checking if such hypotheses already exist, thereby making the tactic idempotent.